user-guide/Advanced_Modules/User Defined APIs/Defining_an_API/Using_existing_scripts.md

```diff
@@ -0,0 +1,166 @@
+
+Imagine we have the following existing script that uses script parameters to pass along the DMA ID and Element ID to stop an element.
+
+```csharp
+using Skyline.DataMiner.Automation;
+
+namespace UDAPIS_Example
+{
+    public class Script
+    {
+        public void Run(Engine engine)
+        {
+            var dmaIdParam = engine.GetScriptParam("DMAID").Value;
+            var elementIdParam = engine.GetScriptParam("ELEMENTID").Value;
+
+            if (!int.TryParse(dmaIdParam, out var dmaId))
+            {
+                engine.ExitFail("Failed to parse DMA ID");
+                return;
+            }
+
+            if (!int.TryParse(elementIdParam, out var elementId))
+            {
+                engine.ExitFail("Failed to parse element ID");
+                return;
+            }
+
+            var element = engine.FindElement(dmaId, elementId);
+            element.Stop();
+        }
+    }
+}
+```
+
+## Option 1: Using the script without the OnApiTrigger entrypoint
+
+To use the above script as an API without doing any changes to it is possible, API triggers will be executed through the `Run` method as if you were executing the script from e.g. Cube. To do this, define your API exactly as explained on the [defining an API page](xref:UD_APIs_Define_New_API#2-creating-the-apidefinitions), but select the `Run` method option as 'Method to be executed'.
+
+Using this method comes with some disadvantages, note that you do not have access to the `ApiTriggerInput` object and `ApiTriggerOutput` object in the script, meaning you cannot check the route or request method of the API trigger or output specific errors.
+
+When the API is triggered with valid input data, the script will succeed and an empty HTTP response will be returned with status code 200. This is an example of a valid input body for the trigger.
+
+```json
+{
+    "DMAID" : "837",
+    "ELEMENTID" : "154"
+}
+```
+
+When the API is triggered with input data that does not represent a number, the conversion in the script will fail and trigger the 'ExitFail' method. This causes the script to fail which will result in an [HTTP response](xref:UD_APIs_Triggering_an_API#errors) like this with status code 500:
+
+```json
+{
+    "errors": [
+        {
+            "title": "Error occurred while handling the request, contact your admin with the provided errorCode and faultingNode ID.",
+            "detail": "Something went wrong, contact the system administrator.",
+            "errorCode": 8,
+            "faultingNode": 837
+        }
+    ]
+}
+```
+
+As you can see, since we do not have a way of returning an `ApiTriggerOutput` instance, the error will be vague. This is why we always recommend spending the little extra time to use the 'OnApiTrigger' entrypoint. (See next option)
+
+## Option 2: Using the script with the OnApiTrigger entrypoint
```

Comment:
Is Option 2 the preferred way? If yes, maybe it should be swapped with option 1, so it's mentioned first? If not, ignore this comment.

Reply:
Option 2 is indeed the preferred way, but the page kind of builds up to first show the easy way where we don't need to do changes and then the extensive (preffered) way where we do make changes in the script. In my opinion, we can put more of an emphasize on the extra functionality there is with the second option.